src/lib.rs

```diff
@@ -236,7 +236,7 @@ impl Map {
     fn new<R: Read>(
         parser: &mut EventReader<R>,
         attrs: Vec<OwnedAttribute>,
-        map_path: Option<&Path>,
+        mut external_file_loader: impl FnMut(&str)->Result<Vec<u8>, TiledError>,
```

Comment:
I'm not sure if this function should return a `TiledError` as `Err`, since it's performing an external data fetch, not something actually related to the Tiled object.

Reply:
Also, maybe instead of `Vec<u8>` it should be an object implementing `Read`, that way implementing async later would be easier.